src/nupic/bindings/engine_internal.i

```diff
@@ -188,6 +192,18 @@
   {
     return nupic::PyArrayRef<nupic::Byte>(self->getOutputData(name)).asNumpyArray();
   }
+
+  // Purge heads of the region's input link buffers
+  void purgeInputLinkBufferHeads()
+  {
+    for (auto & input : self->getInputs())
```

Comment:
No need to change but just a note that if we move the circular buffer into the output in the future then this will need to be changed so that we don't purge multiple times in the case of two inputs linked to the same output.